include/veriblock/entities/vbktx.hpp

```diff
@@ -0,0 +1,109 @@
+#ifndef ALT_INTEGRATION_INCLUDE_VERIBLOCK_ENTITIES_VBKTX_HPP_
+#define ALT_INTEGRATION_INCLUDE_VERIBLOCK_ENTITIES_VBKTX_HPP_
+
+#include <cstdint>
+#include <vector>
+
+#include "veriblock/consts.hpp"
+#include "veriblock/serde.hpp"
+#include "veriblock/slice.hpp"
+
+#include "veriblock/entities/address.hpp"
+#include "veriblock/entities/coin.hpp"
+#include "veriblock/entities/output.hpp"
+#include "veriblock/entities/publication_data.hpp"
+
+namespace VeriBlock {
+
+struct VbkTx {
+  NetworkBytePair networkOrType;
+  Address sourceAddress;
+  Coin sourceAmount;
+  std::vector<Output> outputs;
+  int64_t signatureIndex;
+  PublicationData publicationData;
+  std::vector<uint8_t> signature;
+  std::vector<uint8_t> publicKey;
+
+  VbkTx(NetworkBytePair _networkOrType,
+        Address _sourceAddress,
+        Coin _sourceAmount,
+        std::vector<Output> _outputs,
+        int64_t _signatureIndex,
+        PublicationData _publicationData,
+        Slice<const uint8_t> _signature,
+        Slice<const uint8_t> _publicKey)
+      : networkOrType(std::move(_networkOrType)),
+        sourceAddress(std::move(_sourceAddress)),
+        sourceAmount(_sourceAmount),
+        outputs(std::move(_outputs)),
+        signatureIndex(_signatureIndex),
+        publicationData(std::move(_publicationData)),
+        signature(_signature.begin(), _signature.end()),
+        publicKey(_publicKey.begin(), _publicKey.end()) {}
+
+  static VbkTx fromRaw(ReadStream& stream,
+                       Slice<const uint8_t> _signature,
+                       Slice<const uint8_t> _publicKey) {
+    NetworkBytePair networkOrType = readNetworkByte(stream, TxType::VBK_TX);
+    Address sourceAddress = Address::fromVbkEncoding(stream);
+    Coin sourceAmount = Coin::fromVbkEncoding(stream);
+
+    uint8_t outputSize = stream.readBE<uint8_t>();
+    std::vector<Output> outputs;
+    outputs.reserve(outputSize);
+    for (size_t i = 0; i < outputSize; i++) {
+      outputs.emplace_back(Output::fromVbkEncoding(stream));
+    }
```

Comment:
can be replaced with `readArrayOf`

Reply:
Don't think so. readArrayOf assumes length is stored with writeSingleByteLenValue.